### PR TITLE
ZD3725488 Exit early when app ID is invalid

### PR DESCRIPTION
**Pull request by gfontanilla**
When the app ID provided in `.zat` is invalid and `zat update` is performed, the user gets a `RecordNotFound` (or `Internal Server Error` as reported in the Z1 ticket) error, which is unhelpful.

This PR adds a check to see if the app is installed, and bails early if otherwise.

@zendesk/sustaining @zendesk/vegemite 

### References

* https://support.zendesk.com/agent/tickets/3725488

### Risks

* Low: Might break private app updates via `zat update`

### CODE REVIEW

File: lib/zendesk_apps_tools/deploy.rb [hunk 1]
```diff
@@ -17,6 +17,15 @@ def deploy_app(connection_method, url, body)
       say_error_and_exit e.message
     end
 
+    def check_app_installed(url)
```
Comment (danielbreves) [inline lib/zendesk_apps_tools/deploy.rb]: Considering this method is simply making a GET request with the given url, it doesn't seem to be named appropriately. How about having the API path defined here and only passing the `app_id` as an argument?

File: lib/zendesk_apps_tools/command.rb [hunk 2]
```diff
@@ -191,10 +191,12 @@ def update
       @command = 'Update'
 
       app_id = cache.fetch('app_id') || find_app_id
-      unless /\d+/ =~ app_id.to_s
+      app_url = "/api/v2/apps/#{app_id}.json"
```
Comment (danielbreves) [inline lib/zendesk_apps_tools/command.rb]: This API would indicate whether the app is [available for the account](https://developer.zendesk.com/rest_api/docs/core/apps#get-information-about-app), no? To determine if there's an installation we'd need to GET [`/api/v2/apps/installations/{id}.json`](https://developer.zendesk.com/rest_api/docs/core/apps#show-app-installation).

Comment (gfontanilla) [reply lib/zendesk_apps_tools/command.rb]: @danielbreves 

> This API would indicate whether the app is available for the account, no?

Yes :)

A request to `/api/v2/apps/installations/` and searching for the app ID in the result would work as well. 

If the app is installed, the endpoint `/api/v2/apps/{app_id}.json` should exist/return a 200 (404 if not). The method employed in this PR makes us of this assumption, which I think is quicker (no need to search through a list of IDs).

Comment (fbvilela) [reply lib/zendesk_apps_tools/command.rb]: yes, Daniel is correct here.

File: lib/zendesk_apps_tools/deploy.rb (hunk 1 above)
Comment (fbvilela) [reply lib/zendesk_apps_tools/deploy.rb]: agree, this method will make more sense once it loops through the installations trying to find the `app_id` which should be passed in.

File: lib/zendesk_apps_tools/command.rb (hunk 2 above)
Comment (fbvilela) [reply lib/zendesk_apps_tools/command.rb]: fair enough, if you no longer care if it's installed or not, that endpoint will work 👍

Comment (fbvilela) [reply lib/zendesk_apps_tools/command.rb]: ...as long as you make the request using the account's subdomain and being authenticated, off course

Comment (fbvilela) [reply lib/zendesk_apps_tools/command.rb]: not a caching issue... this endpoint will return the app IF the app is available to you. that's what you want, right? to know if the id exists? What if you are trying to `zat update` an app that hasn't been installed yet? I think you can trust that endpoint. The only problem I see is if you have a wrong id in your .zat file which matches another existing app_id from an app that belongs to another account. In this case, the endpoint will NOT return 200 although the app would exist.

**Pull request merged**